tests/globalvars.cpp

```diff
@@ -0,0 +1,105 @@
+#include "globalvars.h"
+#include "types.h"
+#include "gmock/gmock-matchers.h"
+#include "gtest/gtest.h"
+
+namespace bpftrace::globalvars::test {
+
+using ::testing::HasSubstr;
+
+TEST(GlobalVars, get_named_param_vals)
+{
+  auto global_vars = globalvars::GlobalVars();
+  global_vars.add_named_param("hello", "bye");
+  global_vars.add_named_param("is_true", true);
+  global_vars.add_named_param("will_be_true", false);
+  global_vars.add_named_param("number", 5);
+
+  auto good_values1 = global_vars.get_named_param_vals(
+      { "hello=low", "number=10", "will_be_true" });
+  auto hello_val = good_values1->at("hello");
+  EXPECT_EQ("low", std::get<std::string>(hello_val));
+
+  auto is_true_val = good_values1->at("is_true");
+  EXPECT_EQ(true, std::get<bool>(is_true_val));
+
+  auto is_also_true_val = good_values1->at("will_be_true");
+  EXPECT_EQ(true, std::get<bool>(is_also_true_val));
+
+  auto number_val = good_values1->at("number");
+  EXPECT_EQ(10, std::get<int64_t>(number_val));
+
+  auto good_values2 = global_vars.get_named_param_vals(
+      { "number=-10", "is_true=false" });
+
+  number_val = good_values2->at("number");
+  EXPECT_EQ(-10, std::get<int64_t>(number_val));
+
+  auto is_false_val = good_values2->at("is_true");
+  EXPECT_EQ(false, std::get<bool>(is_false_val));
+
+  auto good_values3 = global_vars.get_named_param_vals({ "number=1m" });
+
+  number_val = good_values3->at("number");
+  EXPECT_EQ(60000000000, std::get<int64_t>(number_val));
+
+  // Test errors
+
+  auto bad_values1 = global_vars.get_named_param_vals({ "random=low" });
+  EXPECT_FALSE(bad_values1);
+  bool has_unknown_opts_error = false;
+  auto ok2 = handleErrors(
+      std::move(bad_values1), [&](const UnknownParamError& uo_err) {
+        EXPECT_THAT(uo_err.err(),
+                    HasSubstr(
+                        "unexpected program command line options: --random"));
+        has_unknown_opts_error = true;
+      });
+  EXPECT_TRUE(has_unknown_opts_error);
```

Comment:
This test pattern repeats multiple times below. How about factoring it out to a function?

Reply:
Done. Well, specifically, the part I refactored was the NamedParamError